OCC.Service.Webhost/Repositories/PersonRepository.cs

```diff
@@ -0,0 +1,125 @@
+
+namespace OCC.Service.Webhost.Repositories
+{
+    public class PersonRepository
+    {
+        private readonly OCCDB _dbContext;
+
+        public PersonRepository(OCCDB dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public int RegisterPerson(Person person)
+        {
+            var p = new OCC.Data.Person();
+            Mapper.CopyProperties(person, p);
+
+            _dbContext.People.Add(p);
+            _dbContext.SaveChanges();
+
+            return p.ID;
+        }
+
+        public Person Login(Person person)
+        {
+            Person dcAttendee = default(Person);
+
+            var bcAttendee =
+                _dbContext.People
+                    .SingleOrDefault(p =>
+                        p.Email == person.Email &&
+                        p.PasswordHash == person.PasswordHash);
+
+            if (bcAttendee != null)
+            {
+                dcAttendee = new Person();
+                Mapper.CopyProperties(bcAttendee, dcAttendee);
+            }
+
+            return dcAttendee;
+        }
+
+        public Person FindPersonByEmail(string email)
+        {
+            Person dcAttendee = default(Person);
+
+            var bcAttendee = _dbContext.People.Where(p => p.Email == email)
+                .SingleOrDefault();
+
+            if (bcAttendee != null)
+            {
+                dcAttendee = new Person();
+                Mapper.CopyProperties(bcAttendee, dcAttendee);
+            }
+            return dcAttendee;
+        }
+
+        public void ResetPassword(string emailAddress, string temporaryPassword, string temporaryPasswordHash)
+        {
+            if (String.IsNullOrEmpty(emailAddress))
+            {
+                throw new ArgumentNullException(emailAddress, "email address must be provided.");
+            }
+
+            var bcAttendee =
+                _dbContext.People.SingleOrDefault(p => p.Email == emailAddress);
+
+            if (bcAttendee == null)
+            {
+                throw new ArgumentOutOfRangeException(emailAddress, "attendee was not found.");
+            }
+
+            if (!String.IsNullOrEmpty(temporaryPasswordHash))
+            {
+                bcAttendee.PasswordHash = temporaryPasswordHash;
+            }
+            _dbContext.SaveChanges();
+
+            IMailService svc = new SmtpMailService();
+            svc.SendPasswordResetMail(emailAddress, temporaryPassword);
+        }
+
+        public void ChangePassword(int id, string oldPasswordHash, string newPasswordHash)
+        {
+            var p = _dbContext.People.Find(id);
+
+            if (String.IsNullOrEmpty(p.PasswordHash))
+            {
+                throw new ArgumentNullException(oldPasswordHash);
+            }
+            if (p.PasswordHash == oldPasswordHash)
+            {
+                p.PasswordHash = newPasswordHash;
+            }
+            _dbContext.SaveChanges();
+
+            IMailService svc = new SmtpMailService();
```

Comment:
In the email I sent separate from the pull request:

"Todo: create test mail service from the IMailService interface"

The plan is to get rid of all of the "new" and move the dependencies into the constructor. I planned on either mocking IMailService or creating a testable implementation of IMailService when adding tests around this repository.